CenterForOpenScience/osf.io
# [Feature] Add common actions menu to component widget [OSF-7442]

## Purpose

Add quick actions menu to component widget on project overview page.
Menu items include:
- Settings link to the component's settings page
- Manage contributors link to the component's contributors page
- Delete option to allow quick delete of the component.

## Changes
- Add html for the menu and css consistent with other osf styles to render_node.mako
- Add component-settings-page.js for click events handling
- Modifies projectSettings.js to allow api url to be passed as argument instead of contextVars
- Modifies project-settings-page.js so it does not break because of changes to projectSettings.js

## Ticket
https://openscience.atlassian.net/browse/OSF-7442

## QA Considerations
Some edge cases QA will want to consider
1. Test deleting both public and private preprints
2. Test that the common actions dropdown button is not visible to people with proper permission
3. Confirm that the delete component button does not show up if the user does not have (admin) permission  to delete the component. 
4. On a registration's component widget, the menu should only have one option (Manage contributors).
5. The button should not show up for non contributors even if parent project is public.
5. Make sure deleting components through the settings page works
6. Make sure nav bar drop down style looks the same (there were some early interactions here)
